glide-core/redis-rs/redis/tests/test_pipeline.rs

```diff
@@ -0,0 +1,589 @@
+
+        // Get the current slot distribution.
+        let cluster_nodes = cluster.get_cluster_nodes().await;
+        let slot_distribution = cluster.get_slots_ranges_distribution(&cluster_nodes);
+        println!("Slot distribution: {:?}", slot_distribution);
+
+        let key = generate_random_string(10);
+        let key2 = generate_random_string(10);
+        let slot_to_move = get_slot(key.as_str().as_bytes());
+
+        // Move the slot of `key` to a different node.
+        cluster
+            .move_specific_slot(slot_to_move, slot_distribution)
+            .await;
+
+        // Create a pipeline with several commands.
+        let mut pipeline = redis::pipe();
+        pipeline.set(&key, "value").get(key).set(key2, "value2");
+
+        // Execute the pipeline.
+        let result = connection
+            .route_pipeline(&pipeline, 0, pipeline.len(), SingleNodeRoutingInfo::Random)
+            .await;
+
+        assert!(
+            result.is_err(),
+            "Expected a MOVED error, but got: {result:?}",
+        );
+        let err = result.unwrap_err();
+        assert_eq!(err.kind(), ErrorKind::Moved, "{err:?}");
+    }
+
+    #[tokio::test]
+    #[serial_test::serial]
+    async fn test_pipeline_read_from_replicas() {
+        // 3 masters, 3 replicas (1 replica per master)
+        let cluster = TestClusterContext::new_with_cluster_client_builder(
+            6,
+            1,
+            |builder| builder.read_from_replicas(),
+            false,
+        );
+
+        // Create an asynchronous connection.
+        let mut connection = cluster.async_connection(None).await;
+
+        // Define pipeline with GET commands.
+        let keys = (1..=15).map(|i| format!("key{}", i));
+        let mut pipeline = redis::pipe();
+        for key in keys {
+            pipeline.get(key);
+        }
+
+        // Route the pipeline explicitly to replicas.
+        let result = connection
+            .route_pipeline(&pipeline, 0, pipeline.len(), SingleNodeRoutingInfo::Random)
+            .await
+            .expect("Pipeline execution failed");
+
+        // Ensure all responses are `Value::Nil`.
+        assert_eq!(result, vec![Value::Nil; pipeline.len()], "{result:?}");
+
+        let mut info_cmd = cmd("INFO");
+        info_cmd.arg("replication").arg("commandstats");
+
+        let res = connection
+            .route_command(
+                &info_cmd,
+                RoutingInfo::MultiNode((MultipleNodeRoutingInfo::AllNodes, None)),
+            )
+            .await
+            .expect("INFO command failed");
+
+        // Parse the INFO output.
+        let info_result: HashMap<String, String> =
+            redis::from_owned_redis_value(res).expect("Failed to parse INFO command result");
+
+        // Aggregate GET command call counts by node role.
+        // Aggregate GET command call counts by node role.
+        let (mut master_get_calls, mut replica_get_calls) = (0, 0);
```

Comment:
minor: looks like can be useful in other use cases, can you move to a separate function something like get_stat_from_info(stat_type (error/command), stat_name (e.g. moved/get))

Reply:
or something that works for your other use cases as well. just seems like it can be handy for future tests. but it isn't a must